src/openforms/js/components/admin/form_design/registrations/zgw/fields/ProductSelect.js

```diff
@@ -0,0 +1,99 @@
+import {useFormikContext} from 'formik';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+import {components} from 'react-select';
+import useAsync from 'react-use/esm/useAsync';
+
+import Field from 'components/admin/forms/Field';
+import FormRow from 'components/admin/forms/FormRow';
+import ReactSelect from 'components/admin/forms/ReactSelect';
+import {get} from 'utils/fetch';
+
+const PRODUCTS_ENDPOINT = '/api/v2/registration/plugins/zgw-api/products';
+
+const getAvailableProducts = async (apiGroupID, catalogueUrl, caseTypeIdentification) => {
+  const response = await get(PRODUCTS_ENDPOINT, {
+    zgw_api_group: apiGroupID,
+    catalogue_url: catalogueUrl,
+    case_type_identification: caseTypeIdentification,
+  });
+  if (!response.ok) {
+    throw new Error('Loading available case type products failed');
+  }
+  return response.data.map(({uri, description}) => ({
+    value: uri,
+    label: uri,
+    description: description,
+  }));
+};
+
+const ProductOption = props => {
+  const {label, description} = props.data;
+  return (
+    <components.Option {...props}>
+      {description && <span className="catalogi-type-option">{description}</span>}
+      <code>{label}</code>
+    </components.Option>
+  );
+};
+
+const ProductSingleValue = props => {
+  const {label, description} = props.data;
+  return (
+    <components.SingleValue {...props}>{description ? description : label}</components.SingleValue>
+  );
+};
+
+const ProductSelect = ({catalogueUrl = ''}) => {
+  const {
+    values: {zgwApiGroup = null, caseTypeIdentification = null},
```

Comment:
```suggestion
    values: {zgwApiGroup = null, caseTypeIdentification = ''},
```
we're not using typescript yet, but still it's good practice to think about the data types :) this field is not nullable, ever.

Reply:
Yeah, you're right. I've fixed it